### ISSUE TITLE
Reconcile fxml/cxml?

### ISSUE THREAD

**Issue by slyrus** [ruricolist/FXML#2]
Hi @ruricolist , there's probably a better place to do this than a github issue, but I'm not sure what it is, so... Here goes. Have you given any thought to trying to reconcile the fxml changes with cxml? It's unfortunate that there are two separately maintained (yes, cxml is at least lightly maintained by the sharplispers group) libraries for dealing with XML.

If not, I might consider using fxml but for the fact that most of my XML usage involves xpath. Why have you chosen not to include plexippus-xpath, or other xpath support, in fxml?

Thanks.

**Comment by ruricolist** [comment]
I took a stab at converting xpath to run on FXML and it went better than I expected. The test suite passes, at least.

The next steps would be to merge the xpath repository into FXML, integrate the xpath test suite with the FXML test suite, and restore xpath compatibility in fxml-stp.

**Comment by slyrus** [comment]
Hmm... While it's nice that xpath works on fxml, I see this as more unnecessary forking.
My preferences would be:
 1) keep the multiple repos, rather than a monorepo
 2) incorporate your fixes (security, etc...) into cxml
 and 3) make a new repo for the integration with the new stuff in fxml.
But, who knows. maybe you're too for down the forking path.

**Comment by ruricolist** [comment]
For several years I maintained a repository that compiled all of the available bugfixes for CXML. I even wrote offering to become the new maintainer, but never received an answer.

I only decided to fork when I found out how insecure CXML was, and realized that addressing that would require significant changes to the API. (It's possible to use FXML is a mostly CXML-compatible way, but the defaults are necessarily different.)

Since then I've been trying to tame this huge, ancient, largely undocumented beast, to simplify, rationalize, document and test as much as possible to make it humanly maintainable in the long term.

I wish CXML well. I will go through FXML and backport any pure bug fixes. I'll even extract the CXML test suite, which was a pain to get working, so you all have something to test against. But in the long run I think FXML is a better approach.
